lib/hammer_cli_foreman/hosts/common_update_options.rb

```diff
@@ -76,10 +76,16 @@ def request_params
         params['host']['enabled'] = option_enabled unless option_enabled.nil?
         params['host']['overwrite'] = option_overwrite unless option_overwrite.nil?
 
-        params['host']['host_parameters_attributes'] = parameter_attributes
+        params['host']['host_parameters_attributes'] = parameter_attributes unless option_parameters.nil?
         params['host']['compute_attributes'] = option_compute_attributes || {}
-        params['host']['compute_attributes']['volumes_attributes'] = nested_attributes(option_volume_list)
-        params['host']['interfaces_attributes'] = interfaces_attributes
+
+        if action == :update
+          params['host']['compute_attributes']['volumes_attributes'] = nested_attributes(option_volume_list) unless option_volume_list.empty?
+          params['host']['interfaces_attributes'] = interfaces_attributes unless option_interface_list unless option_interface_list.empty?
+        else
+          params['host']['compute_attributes']['volumes_attributes'] = nested_attributes(option_volume_list)
+          params['host']['interfaces_attributes'] = interfaces_attributes unless option_interface_list
```

Comment:
As I said in my previous comment, `option_interface_list` is always true and that makes this condition to be always false -> the attributes will be nil. The tests are actually failing for that right now.